lib/OpenQA/Worker/Job.pm

```diff
@@ -1077,14 +1078,16 @@ sub _upload_asset {
                 local => $local_upload,
                 retries => $retries
             });
-    };
-    log_error($@, channels => \@channels_both, default => 1) if $@;
-
+    }
+    catch ($e) {
+        $error = $e;
```

Comment:
I'm sorry, but after thinking about it what would make it the same logic as before would be
```
$error ||= $e;
```